src/services/services.ts

```diff
@@ -6099,6 +6197,25 @@ namespace ts {
                 const start = findInComments ? container.getFullStart() : container.getStart();
                 const possiblePositions = getPossibleSymbolReferencePositions(sourceFile, searchText, start, container.getEnd());
 
+                // If we are just looking for implementations and this is a property access expression, we need to get the
+                // symbol of the local type of the symbol the property is being accessed on. This is because our search
+                // symbol may have a different parent symbol if the local type's symbol does not declare the property
+                // being accessed (i.e. it is declared in some parent class or interface)
+                let parents: Symbol[];
+                const cache: Map<boolean> = createMap<boolean>();

```

Comment:
This should probably be called `inheritsFromCache`.
Also, `SymbolInheritanceState` has its own `cachedInheritanceResults`, but that doesn't seem to be used.